fixed create region api route bug

peichins opened the pull request:
# Title of PR

Fixed create region api route bug

## Changes

Changed the 2nd parameter on the region service create function call to use the project object rather than the (missing) project id property on the form model.

Also added a test for the value of this 2nd parameter. 

## Problems

Any problems caused by the PR that will need to be address in another PR. For example, these issues may be caused by another branch which will change how this branch functions.

## Issues

fixes #1985 

## Visual Changes

none

## Final Checklist

- [x] Assign reviewers if you have permission
- [x] Assign labels if you have permission
- [x] Link issues related to PR
- [x] Ensure project linter is not producing any warnings (`npm run lint`)
- [ ] Ensure build is passing on all browsers (`npm run test:all`)
- [ ] Ensure CI build is passing
- [ ] Ensure docker container is passing ([docs](https://github.com/QutEcoacoustics/workbench-client#docker))

peichins committed:
fixed create region api route

atruskie reviewed (commented):
Looks like a nice neat change, however, I have questions:

1. Why is the project resolver not required in either of the `linkToRoute` functions - as it is in https://github.com/QutEcoacoustics/workbench-client/blob/ad1c1aff51831d12f5c767e100fe2c4a9ba2c087/src/app/components/sites/pages/new/new.component.ts#L77-L80
2. Did you test the shallow route?

peichins committed:
started fixing problem with resolver for create region on shallow route

atruskie committed:
Fixed registration of default project resolver

I actually extended the resolvers class just to make all the resolvers sit in one place. Looks  nicer.

atruskie committed:
StrongRoute props now hidden behind getters

Previously, StrongRoute props were public. Because those props encoded both parent and child relationships, iterating over those props in an object  graph walk would result in cyclical loops.

JSON.stringify could not serialize StrongRoute before.
More importantly ngMocks would fail when used with any PageInfo object - that made mocking any of our components with our standard route information impossible.

It seems by using getters (and ECMA private fields) both the above cases no longer occur.

atruskie committed:
Wrote a unit test for new regions resolvers!

Wrote our fest ever test for route resovlers.

Worthwhile doing because we had some engineering blocks and the regions routes used two different resolvers.
